### PR TITLE
refactor: simplify observer response for resource deployment

### PR DESCRIPTION
**Pull request by deryrahman**

### CODE REVIEW

File: core/sender/progress_count_deploy_resource.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package sender
+
+import (
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus/core/v1beta1"
+)
+
+type deployResourceProgressCount struct {
```
Comment (sravankorumilli) [inline core/sender/progress_count_deploy_resource.go:7]: We don't need the progress, as this is not helpful to the users and this response is tightly coupled with the optimus cli. So, better avoid it.

File: core/sender/log_status_deploy_resource.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package sender
+
+import (
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus/core/v1beta1"
+)
+
+type deployResourceLogStatus struct {
```
Comment (sravankorumilli) [inline core/sender/log_status_deploy_resource.go:7]: Similar to job deploymentstatus we can do the same here

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

File: cmd/deploy/deploy.go [hunk 3]
```diff
@@ -313,13 +313,11 @@ func (d *deployCommand) processResourceDeploymentResponse(
...
 ) error {
 	d.logger.Info("> Receiving responses:")
-	var counter int
 	spinner := progressbar.NewProgressBar()
 	defer spinner.Stop()
 
-	if !d.verbose {
-		spinner.StartProgress(totalSpecsCount, "please wait")
-	}
+	spinner.StartProgress(totalSpecsCount, "please wait")
```
Comment (sravankorumilli) [inline cmd/deploy/deploy.go:319]: we don't need this right?

File: cmd/deploy/deploy.go [hunk 4]
```diff
@@ -328,19 +326,21 @@ func (d *deployCommand) processResourceDeploymentResponse(
 			}
 			return err
 		}
-		if resp.GetAck() {
-			if !resp.GetSuccess() {
-				d.logger.Error(resp.GetMessage())
+
+		if logStatus := resp.GetLogStatus(); logStatus != nil {
```
Comment (sravankorumilli) [inline cmd/deploy/deploy.go:330]: we can move the log handling to a single place, this is duplicate code in all the commands

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

File: cmd/deploy/deploy.go (hunk 3 above)
Comment (deryrahman) [reply cmd/deploy/deploy.go:319]: yes, we don't need it anymore. deleted

File: cmd/deploy/deploy.go (hunk 4 above)
Comment (deryrahman) [reply cmd/deploy/deploy.go:330]: let's doing it once it's merged on `simplify-observer-response` branch

**Pull request merged**